File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.py
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        
        players = [1]*n
        eleminated = 0
        
        i = 0
        while True:
            cnt = k-1
            while (cnt>0 or players[i]<0):
                
                if(players[i] < 0):
                    i += 1
                    i = i%n
                    continue
                    
                cnt -= 1
                i += 1
                i = i%n
            
            if(eleminated == n-1):
                return i+1
            
            players[i] = -1
            eleminated += 1
            i += 1
            i = i%n

File: 1823-find-the-winner-of-the-circular-game/1823-find-the-winner-of-the-circular-game.cpp
class Solution {
public:
    int findTheWinner(int n, int k) {
        vector<int> players(n, 1);
        int eleminated = 0;

        int i = 0;

        while(true){
            int cnt = k-1;
            while(cnt>0 || players[i]<0){
                if(players[i] < 0) {
                    i++;
                    i = i%n;
                    continue;
                }
                cnt--;
                i++;
                i = i%n;
            }
            if(eleminated == n-1)   return i+1;
            players[i] = -1;
            eleminated++;
            i++;
            i = i%n;
        }
    }
};